### PR TITLE
Initial draft of corporate engagement guide.

### PR DESCRIPTION
**Pull request by ellisonbg**
This is an initial draft of a corporate engagement guide. I have tried to summarize the best practices and principles that have emerged from our collaborations with companies (Bloomberg, TwoSigma, JPM, Anaconda, Quansight, etc.). Would love comments!

### CODE REVIEW

File: jupyterlab/CORPORATE.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+the project, everyone contributes as individuals. Because of this, if your
+corporation wants to engage with and contribute to JupyterLab, the primary means
+will be through allocating, empowering, and encouraging your individual
+employees to contribute in an ongoing manner. Put more bluntly, pay your
+employees to contribute to Jupyter.
+
+**Make balanced contributions.** We expect all contributors to make balanced
+contributions that match their skill level and level of participation. Balanced
+contributions include work on new features, fixing bugs, reviewing pull
+requests, writing documentation, issue triage, answering users' questions,
```
Comment (afshin) [inline jupyterlab/CORPORATE.md]: > answering users' questions

This is a really good thing to specifically enumerate. Answer questions on channels like Gitter helps expand community.

File: jupyterlab/CORPORATE.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+**Hire our core maintainers and give them more time to contribute.** Our core
+maintainers love working on Jupyter, but have to balance those contributions
+with the reality of paid employment. If you are interested in hiring a core
+maintainer, we encourage you to do this in a way that gives them **more** time
+to contribute than they currently have. Put this into their employment contract.
+This is an incredibly effective way of turbocharging the work of the project.
+Conversely, if you hire a maintainer and give them less time to contribute, you
+are actively damaging the project and your organization will likely find itself
+alienated from the community.
```
Comment (saulshanabrook) [inline jupyterlab/CORPORATE.md]: "alienated" seems a little harsh here.

**Comment by scopatz** [comment]
Hi @ellisonbg - thanks for putting this together!  I think it is well-structured and representative of how JupyterLab feels.  I have a couple of suggestions for improvements, though.

1. Be more explicit on how companies can engage
2. Reinforce the puppy metaphor

### You get it or you don't

I think people and corporate entities who are used to participating in open source, community-lead projects before will totally understand this document.  However, for people and companies that are unfamilar with this kind of development, I was hoping for more process details for activities things like:

* Funding a call for support (ie JupyterLab puts out a call to fund X new feature or maintain Y, where does company A go to fund that)
* How do the funding calls get issued (can be a link to an external document), and where can I make suggestions
* Who do I contact if I want to hire members of the core team
* How do I donate to the project if I don't care about a specific SOW

### Puppy Metaphor

I am a fan, but I think it could be better reinforced in the following parts of the document. Maybe there could be a brief, italicizied sentence that relates how each point fits in the mental model. For instance, 

* Empower individual contributors - Encourage your empolyees to help at the puppy yard.
* Make balanced contributions - You can help take care of the puppies in a variety of ways (feeding walking, grooming, medical care etc).
* Help review pull requests - If you help scoop up after other puppy, we're more likely to help when your favorite puppy needs attention
* Use public communications channels whenever possible - Be sure to check and post to bulletin board for the latest updates about the puppy yards
* Hire our core maintainers and give them more time to contribute - consider paying the regulars around here to keep looking after the puppies.
* Commit to maintaining your contributions - Don't abbandon puppies here!
* Don't surprise us - Don't build us a new dog park when what we need is just a couple of more kennels.
* Be patient - Puppies require training, and complex new tricks can take years and the right motivation to learn.

--- 

Thanks so much for writing this up @ellisonbg!

File: jupyterlab/CORPORATE.md (hunk 2 above)
Comment (sccolbert) [reply jupyterlab/CORPORATE.md]: I think it's appropriately harsh.

File: jupyterlab/CORPORATE.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+channels above). See if there are existing issues on the topic, or open a new
+issue describing the work. There may be other people already working on it, or
+the work may be blocked by other factors.
+
+**Be patient.** Open-source software projects offer a unique path to innovation
+by bringing together users and contributors with diverse needs and usage cases.
+Even if fully resourced, the resulting innovation can be slow, possibly much
+slower than your corporation's desired timeline. Successful open-source engagement
+will require long term thinking and strategy. Be patient with us, set
+appropriate expectations, and find ways to mitigate the risk.
```
[the remark this answers is not in the document]
Comment (ellisonbg) [reply jupyterlab/CORPORATE.md]: Thanks for the suggestion - working on integrating it.

**Comment by ellisonbg** [comment]
@scopatz - thanks for the details comment and suggestions. I have incorporated many of them. For now, I am not trying to tackle the bigger questions of funding, roadmaps, etc. But I agree that stuff is related and also important.

I have softened the language a bit on the employment questions and clarified that underlying assumptions (that our language only applies when maintainers want to work on the project - if they want to work on other things, that is totally fine). Let me know if this needs further tuning.

**Comment by ellisonbg** [comment]
I think all feedback has been addressed. Can we merge?

**Comment by blink1073** [comment]
In she goes, thanks!

File: jupyterlab/CORPORATE.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+amazing!!! Lots of people stop by to look at the puppies, play with them, take
+them on walks, and play fetch. These people are like the users of an open-source
+project. A few people, the maintainers, live in the yard with the puppies. The
+maintainers feed the puppies, take care of them, wash them, take them to the
+vet, get their shots, and keep the yard clean. Just to keep the existing puppies
+happy and healthy requires a ton of work and money.
+
+Working in a yard full of puppies is CRAZY! At times, there can be a line around
+the block of people wanting to play with the puppies and talk, or complain, to
+the maintainers. Sometimes, people show up wth more puppies. These are like pull
```
Comment (aterrel) [inline jupyterlab/CORPORATE.md]: `people show up wth more puppies` -> `people show up with more puppies`

**Pull request merged**